pkg/kube/client.go

```diff
@@ -610,6 +644,53 @@ func (c *client) GetIstioVersions(ctx context.Context, namespace string) (*versi
 	return &res, errs
 }
 
+func (c *client) getIstioVersionUsingExec(pod *v1.Pod) (*version.BuildInfo, error) {
+
+	// exclude data plane components from control plane list
+	labelToPodDetail := map[string]struct {
+		binary    string
+		container string
+	}{
+		"pilot":            {"/usr/local/bin/pilot-discovery", "discovery"},
+		"istiod":           {"/usr/local/bin/pilot-discovery", "discovery"},
+		"citadel":          {"/usr/local/bin/istio_ca", "citadel"},
+		"galley":           {"/usr/local/bin/galley", "galley"},
+		"telemetry":        {"/usr/local/bin/mixs", "mixer"},
+		"policy":           {"/usr/local/bin/mixs", "mixer"},
+		"sidecar-injector": {"/usr/local/bin/sidecar-injector", "sidecar-injector-webhook"},
+	}
+
+	component := pod.Labels["istio"]
+
+	// Special cases
+	switch component {
+	case "statsd-prom-bridge":
+		// statsd-prom-bridge doesn't support version
+		return nil, fmt.Errorf("statsd-prom-bridge doesn't support version")
+	case "mixer":
```

Comment:
I think this `case` is no longer needed as `mixer` is completely removed for `1.8`?

Reply:
I left this in because I want to cherry-pick to 1.7, and because 1.8 _istioctl_ should do best effort against 1.7 control planes.